scripts/addr2line.py

```diff
@@ -83,41 +84,148 @@ def __call__(self, address):
         self._input.stdin.flush()
         return self._read_resolved_address()
 
+class KernelResolver:
+    """A reserver for kernel addresses which tries to read from /proc/kallsyms."""
+
+    LAST_SYMBOL_MAX_SIZE = 1024
+
+    def __init__(self, binary, *args, **kwargs):
+        assert binary == '<kernel>', 'this resolver only makes sense for the kernel'
+        syms : list[tuple[int, str]] = []
+
+        ksym_re = re.compile(r'(?P<addr>[0-9a-f]+) (?P<type>.+) (?P<name>\S+)')
+
+        warnings_left = 10
+        try:
+            f = open('/proc/kallsyms', 'r')
+        except OSError as e:
+            self.error = f'Cannot open /proc/kallsyms: {e}'
+            print(self.error)
```

Comment:
Not sure what the convention is in this tool for logging, should this error be sent to stderr rather than stdout?

Reply:
All explicit output goes to stdout in this tool, including errors (there may be some "implicit" output, e.g., that output by subprocesses like addr2line which ends up on stdout). So I think it should go to stdout, especially given that this is a soft error (we just do not resolve the addresses).